python/python/lance/dataset.py

```diff
@@ -1097,38 +1097,6 @@ def _to_inner(self):
                 raw_updated_fragments, self.deleted_fragment_ids, self.predicate
             )
 
-    @dataclass
-    class Rewrite(BaseOperation):
```

Comment:
Yes.  I talk about it a bit in the description:

> I also removed the "rewrite operation" from the python bindings. This was already a fairly complicated operation and adding rewritten indices to it would make it even more complicated. I don't think users are going to be orchestrating compaction themselves anytime soon (and we can add it back in if needed). Furthermore, the plan compaction capability and serialization of compaction tasks is a much better solution for anyone that wants to orchestrate the compaction process.

There's no reason we can't support it but we'd need to fix that test and I worry it'll be too hard to understand at the moment.

Reply:
That's fair. I don't mind not supporting it. We still have another pathway for distributed compaction, right?